src/magic_folder/config.py

```diff
@@ -222,6 +240,9 @@ def create_global_configuration(basedir, api_endpoint_str, tahoe_node_directory,
         database=connection,
         api_token_path=basedir.child(b"api_token"),
     )
+    # check that these are valid by setting them
+    config.api_endpoint = api_endpoint_str
+    config.api_client_endpoint = api_client_endpoint_str
```

Comment:
It would be cool if we *only* inserted values into the database through a Python setter interface (ie, if we didn't need the `INSERT INTO` SQL statement a few lines above this code).

A runner-up might be if there were an explicit validation API that could be invoked separate from the setter so we don't need to invoke the setter just for the validation side-effect.

Reply:
Might just be my lack of SQL/SQLite understanding .. I believe my thinking there is that the `INSERT INTO` above is to put a row in so that the setters can always just `UPDATE`.

But, okay so yeah if I move the validators separately we can call them here before the INSERT (and call the same ones in the setters).